### PR TITLE
AK/clean up server file endpoints

### PR DESCRIPTION
**Pull request by RyuGuardian**

### CODE REVIEW

File: server/app.py [hunk 1]
```diff
@@ -220,102 +131,32 @@ def whosHereAPI():
...
+        return parser_page(nav.get_filepath_for_endpoint(endpoint))
+      else:
+        abort(404)
 
-    @app.route("/docs/skills")
-    def docs_skills():
-        return parser_page('skills_filepath')
+    @app.route('/items/<category>')
+    def items(category):
+      endpoint = '/items/' + category
```
Comment (bleehu) [inline server/app.py]: I love this! So much cleaner!

File: server/app.py [hunk 2]
```diff
@@ -435,11 +272,6 @@ def missed_it(error):
         seconds_out = config.get('WhosHere', 'Seconds_out')
     guestbook.initialize(seconds_away, seconds_out)
 
-    local_dir = os.path.dirname(__file__) #get local directory, so we know where we are saving files.
-    log_filename = os.path.join(local_dir,"cxDocs.log") #save a log of web traffic in case something goes wrong.
-    logging.basicConfig(filename=log_filename, level=logging.INFO)
-    global log
-    log = logging.getLogger("cxDocs:")
```
Comment (bleehu) [inline server/app.py]: Why did we move this to the top of the file?

File: server/navigation/_manual_routes.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# Gives the initial piece of the dictionary where we have to manually enter routes and endpoints
+def get_manual_routes():
```
Comment (bleehu) [inline server/navigation/_manual_routes.py]: rather than hard-code this, would it make sense to make this a config file? Probably in .json format?

File: server/navigation/nav_dict.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        rules_dict = _rules.generate_all_routes_dict(options_list)
+
+        if rules_dict != None and len(rules_dict) > 0:
+            _nav_dict.update(rules_dict)
+
+        if len(_nav_dict) == length_before_docs:
+            print("Looked for documents, but didn't find any. See config/README.md to configure rules docs.")
+
+    else:
+        print("Error: _nav_dict not created.")
```
Comment (bleehu) [inline server/navigation/nav_dict.py]: I would kinda like to see error handling logged, but asking that would scope creep.

File: server/navigation/nav_dict.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    for route in _nav_dict[endpoint]['navbar']:
+        list_results.append([])
+
+        if 'nav_list' in _nav_dict[route]:
+            for item in _nav_dict[route]['nav_list']:
+                list_results[i].append( (_nav_dict[item]['label'], item) )
+
+        i += 1
+
+    return list_results
```
Comment (bleehu) [inline server/navigation/nav_dict.py]: your code in this file especially is organized very well by level of abstraction. I was able to read it and understand it by simply going down the page. Great example!

File: server/navigation/_rules.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            }
+
+            dict_length -= 1
+
+    if dict_length > 0:
+        print("Warning: Discrepancy between {} rules paths and config options.".format(category))
+
+    if len(route_dict) < 1:
+        print("Looked for {} rules documents, but didn't find any. See config/README.md to configure rules docs.".format(category))
+        return None
```
Comment (bleehu) [inline server/navigation/_rules.py]: A little leery about returning None here instead of handling an error.

File: server/templates/home.html [hunk 7]
```diff
@@ -4,92 +4,79 @@
...
-        {{ list.holoListOfLinks(character_docs) }}
 
-      </div>
-      {% endif %}
+  {# MEMBERS SECTION #}
+  {% if 'username' in session %}
 
-      {% if itemsDocs != None %}
-      <div class="col-lg-3 col-md-3 col-sm-4">
+  {# Not sure how many of these are needed. For now, they are at most only for members. #}
```
Comment (bleehu) [inline server/templates/home.html]: if you're not sure, ask on slack!
It would be nice if the NPC generator and Files page were available without logging in. The files page has our old character sheet, and hopefully will have our new one soon. Not everyone who plays CX has a login yet. You're right, it would be nice to have a register button.

File: server/navigation/nav_dict.py (hunk 4 above)
Comment (RyuGuardian) [reply server/navigation/nav_dict.py]: Yeah, I'm wary about passing the logger to everything, but that may be the Python way? I could add it if you want. I was thinking of it, but decided to submit what I had so far as it could be addressed later.

File: server/navigation/nav_dict.py (hunk 5 above)
Comment (RyuGuardian) [reply server/navigation/nav_dict.py]: Thanks!

File: server/navigation/_rules.py (hunk 6 above)
Comment (RyuGuardian) [reply server/navigation/_rules.py]: TBF, this is basically copied code (I just changed the conditional and the print string slightly). As I'm used to JS and front-end work, I'd rather the website still be viewable if there's a small problem gathering data than show a debugging/error page. We could change this if you really want, though.

File: server/templates/home.html (hunk 7 above)
Comment (RyuGuardian) [reply server/templates/home.html]: There's a lot of uglifying of the front end before it gets prettier, so sometimes I just put things aside to worry about later, and I'm trying to organize according to the logic in my head and forget that players are actually using this site.

Players without accounts can still access /files and /npcgen, there just isn't explicit navigation.
